test/reshape.jl

```diff
@@ -507,4 +507,45 @@ end
     @test eltype(typeof(sdf2.value)) === Float64
 end
 
+@testset "transpose" begin
+    df1 = DataFrame(a=["x", "y"], b=rand(2), c=[1,2], d=rand(Bool,2))
+    df2 = DataFrame(a=["x", "y"], b=[1., "str"], c=[1,2], d=rand(Bool,2))
+    df3 = DataFrame(a=fill("x", 10), b=rand(10), c=rand(Int, 10), d=rand(Bool,10))
```

Comment:
please check:
1) `DataFrame()`
2) `DataFrame(a=Int[], b=Float64[])`
3) and cases that should error (e.g. missing column, column with new names that is not `Symbol` or string etc.)

Reply:
Just to be sure, when would expect:

1. `permutedims(DataFrame()) == DataFrame()` or error?
2. I'm thinking error, since there are no values for headers?
3. when you say "missing column", do you mean where the `src_namescol` has a `missing`? Or where certain columns have `missing` values?

Reply:
Ad.1 - it should error I think
Ad 3. - when I say "missing column" I mean "when the requested column is not present in a data frame"